native/cocos/bindings/manual/jsb_cocos_manual.cpp

```diff
@@ -689,44 +689,57 @@ static bool register_se_setExceptionCallback(se::Object *obj) { // NOLINT(readab
     return true;
 }
 
-static bool js_engine_Color_get_val(se::State &s) // NOLINT(readability-identifier-naming)
+static bool js_engine_Color_get_data(se::State &s) // NOLINT(readability-identifier-naming)
 {
     auto *cobj = SE_THIS_OBJECT<cc::Color>(s);
     SE_PRECONDITION2(cobj, false, "Invalid Native Object");
 
     CC_UNUSED bool ok = true;
     se::Value jsret;
-    auto r = static_cast<uint32_t>(cobj->r);
-    auto g = static_cast<uint32_t>(cobj->g);
-    auto b = static_cast<uint32_t>(cobj->b);
-    auto a = static_cast<uint32_t>(cobj->a);
-    uint32_t val = (a << 24) + (b << 16) + (g << 8) + r;
-    ok &= nativevalue_to_se(val, jsret, s.thisObject() /*ctx*/);
-    s.rval() = jsret;
+    uint8_t data[4] = {
+        static_cast<uint8_t>(cobj->r),
+        static_cast<uint8_t>(cobj->g),
+        static_cast<uint8_t>(cobj->b),
+        static_cast<uint8_t>(cobj->a)};
+    se::Object *dataObj = se::Object::createTypedArray(se::Object::TypedArrayType::UINT8_CLAMPED,
```

Comment:
Use se::HandleObject for non-jsb objects to avoid memory leak and gc issues.
```cpp
se::HandleObject dataObj(se::Object::createTypedArray(...));
```

Reply:
fixed